### DISCUSSION TITLE
Episode 180: The C++0x Concepts Story with Doug Gregor (Part 1)

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#72] [Episode]
Episode Link: https://adspthepodcast.com/2024/05/03/Episode-180.html

Feel free to write comments or ask questions 🙂

**Comment by JVApen** [comment]
If I understood correctly, the big differences between C++0x and C++20 are:
 - No guarantee that the template is constraint to the given contract requirements 
 - No mapping between classes and the concepts they implement

The first seems something that easily can be checked by static analysis. Does this already exist or is someone working on this?
The second might be more complex for static analysis, does it seem possible to write that?

**Reply by WimLeflere** [reply]
Having the compiler check the contract for completeness should be feasible (new warning to be added?), I think.
